### PR TITLE
ccc twenty four add recursion and java+python sols

### LINKED ISSUE #3422: ccc '24's internal sol doesn't use recursion
[this problem](https://usaco.guide/bronze/complete-rec#problem-ccc-Twenty-four), even though it's in the complete search w/ rec module,,,, doesn't have an internal solution that uses recursion

### PR DESCRIPTION
**Pull request by 876pol**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.
 
 fixes #3422

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
nonononono

**Comment by SansPapyrus683** [comment]
@876pol any updates?

**Comment by 876pol** [comment]
oh oops kinda forgot about this

File: solutions/bronze/ccc-Twenty-four.mdx [hunk 1]
```diff
@@ -44,77 +47,322 @@ int operation(int op, int num1, int num2) {
...
+void generate_hand_permutation() {
+	if (hand_permutation.size() == 4) {
+		// We have generated a permutation, so we can try placing the
+		// operators.
+		for (int op1 = 0; op1 < 4; op1++) {
+			for (int op2 = 0; op2 < 4; op2++) {
+				for (int op3 = 0; op3 < 4; op3++) {
+					int first = operation(op1, hand_permutation[0],
+										  hand_permutation[1]);
+					// If the operation is invalid, continue;
```
[the remark this answers is not in the document]
Comment (876pol) [reply solutions/bronze/ccc-Twenty-four.mdx:60]: the indent spacing is different on github
the alignment's better on usaco guide: [](https://usaco-guide-flykvef9w-cpinitiative.vercel.app/problems/ccc-twenty-four/solution)

[the remark this answers is not in the document]
Comment (SansPapyrus683) [reply solutions/bronze/ccc-Twenty-four.mdx:60]: i'd rather you indent it like
```cpp
int x = really_long_function(
	asdf,
	asdf,
);
```

[the remark this answers is not in the document]
Comment (876pol) [reply solutions/bronze/ccc-Twenty-four.mdx:60]: ok i have changed it

File: solutions/bronze/ccc-Twenty-four.mdx [hunk 2]
```diff
@@ -44,77 +47,383 @@ int operation(int op, int num1, int num2) {
...
+	}
+}
+
 int main() {
 	int num_hands;
 	cin >> num_hands;
 
 	for (int h = 0; h < num_hands; h++) {
-		vector<int> hand(4);
+		ans = -INF;
```
Comment (SansPapyrus683) [inline solutions/bronze/ccc-Twenty-four.mdx:148]: replace INF with INT32_MIN

File: solutions/bronze/ccc-Twenty-four.mdx [hunk 3]
```diff
@@ -44,77 +47,383 @@ int operation(int op, int num1, int num2) {
...
+</CPPSection>
+
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class Solution {
+	static final int INF = (int) 1e9;
```
Comment (SansPapyrus683) [inline solutions/bronze/ccc-Twenty-four.mdx:169]: same here, but w/ Integer.MIN_VALUE

File: solutions/bronze/ccc-Twenty-four.mdx [hunk 4]
```diff
@@ -44,77 +47,383 @@ int operation(int op, int num1, int num2) {
...
 }
 ```
 
-</CPPSection>
+</JavaSection>
+
+
+<PySection>
+```py
+INF = int(1e9)
```
Comment (SansPapyrus683) [inline solutions/bronze/ccc-Twenty-four.mdx:317]: python has `float('inf')`

File: solutions/bronze/ccc-Twenty-four.mdx [hunk 5]
```diff
@@ -44,77 +47,383 @@ int operation(int op, int num1, int num2) {
...
+				continue
+			chosen[i] = True
+			hand_permutation.append(hand[i])
+			generate_hand_permutation()
+			chosen[i] = False
+			hand_permutation.pop()
+
+num_hands = int(input())
+
+for h in range(num_hands):
```
Comment (SansPapyrus683) [inline solutions/bronze/ccc-Twenty-four.mdx:417]:
```suggestion
for _ in range(int(input())):
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/bronze/ccc-Twenty-four.mdx [hunk 6]
```diff
@@ -314,7 +312,7 @@ public class Solution {
 
 <PySection>
 ```py
-INF = int(1e9)
+INF = float('inf')
```
Comment (SansPapyrus683) [inline solutions/bronze/ccc-Twenty-four.mdx:315]: just remove this variable at this point lmao

Comment (SansPapyrus683) [reply solutions/bronze/ccc-Twenty-four.mdx:315]: bump

**Comment by SansPapyrus683** [comment]
could you add a time complexity?

**Comment by 876pol** [comment]
the code still works

however, the pre-commit.ci task (what i assume to be an auto-formatter) changed the way the file was formatted previously. should i/is it possible to change it back?

**Comment by SansPapyrus683** [comment]
unfortunately no

**Pull request merged**